api/serializers.py

```diff
@@ -0,0 +1,108 @@
+from rest_framework import serializers
+from rest_framework.pagination import PageNumberPagination
+
+from game.models import Game, Genre, Platform, Screenshot, Cover, UserFavouriteGame
+from django.conf import settings
+
+
+class MainPageGamesSetPagination(PageNumberPagination):
+    """Sets games limit per page displayed on Main Page"""
+
+    page_size = settings.GAMES_PER_PAGE_MAIN
+
+
+class MustsPageGamesSetPagination(PageNumberPagination):
+    """Sets games limit per page displayed on Musts Page"""
+
+    page_size = settings.GAMES_PER_PAGE_MUSTS
+
+
+class GameSerializer(serializers.ModelSerializer):
+    """Serializer for Game Model"""
+
+    class Meta:
+        model = Game
+        fields = '__all__'
+
+    def to_representation(self, instance):
+        """Ensures only not null keys are displayed"""
+
+        result = super(GameSerializer, self).to_representation(instance)
+        return {key: result[key] for key in result if result[key]}
+
+
+class GenreSerializer(serializers.ModelSerializer):
+    """Serializer for Genre Model"""
+
+    class Meta:
+        model = Genre
+        fields = '__all__'
+
+
+class PlatformSerializer(serializers.ModelSerializer):
+    """Serializer for Platform Model"""
+
+    class Meta:
+        model = Platform
+        fields = '__all__'
+
+    def to_representation(self, instance):
+        """Ensures only not null keys are displayed"""
+
+        result = super(PlatformSerializer, self).to_representation(instance)
+        return {key: result[key] for key in result if result[key]}
+
+
+class ScreenshotSerializer(serializers.ModelSerializer):
+    """Serializer for Screenshot Model"""
+
+    class Meta:
+        model = Screenshot
+        fields = '__all__'
+
+
+class CoverSerializer(serializers.ModelSerializer):
+    """Serializer for Cover Model"""
+
+    class Meta:
+        model = Cover
+        fields = '__all__'
+
+
+class FavouritesPostSerializer(serializers.Serializer):
+    """
+    Serializer created to validate POST method body for FavouritesAPIView. Takes {action} and {game_ids} values from
+    POST request and checks their correctness.
+    """
+
+    action = serializers.ChoiceField(choices=["add", "delete"])
+    game_ids = serializers.ListField()
+
+    def validate(self, data):
```

Comment:
if you're validating game_ids only, it's better to define field validator: `def validate_game_ids(self, data)`. `validate()` is used for complex validation cases where multiple fields are involved

Reply:
Tried to define field validator, but my <code>game_ids</code> validation depends on the <code>action</code>, so some initial check for this field needed. In this case, <code>validate()</code> works good for me: I can get all the data and perform cross-field validation.

Reply:
ohh, I see, sorry for the confusion - all good